spec/services/generate_test_data_spec.rb

```diff
@@ -22,5 +22,13 @@
         expect(application_choice.provider.code).to eq 'DEF'
       end
     end
+
+    it 'works when duplicate course codes are generated' do
```

Comment:
the description `it works when ...` it's a bit generic, maybe something like 
`generates unique course code`?